include/core/CCompressedLfuCache.h

```diff
@@ -453,15 +469,17 @@ class CCompressedLfuCache {
     }
 
     bool full(std::size_t itemMemoryUsage) const {
-        double memory{static_cast<double>(this->unguardedMemoryUsage() + itemMemoryUsage)};
+        std::size_t memory{this->unguardedMemoryUsage() + itemMemoryUsage +
+                           sizeof(typename TCompressedKeyCacheItemUMap::value_type) +
+                           sizeof(typename TCacheItemStatsSet::value_type)};
         if (this->needToResizeItemCache()) {
-            memory += (static_cast<double>(this->nextItemCacheBucketCount()) /
-                           static_cast<double>(m_ItemCache.bucket_count()) -
-                       1.0) *
-                      static_cast<double>(core::CMemory::dynamicSize(m_ItemCache));
+            memory += static_cast<std::size_t>(
+                (static_cast<double>(this->nextItemCacheBucketCount()) /
+                     static_cast<double>(m_ItemCache.bucket_count()) -
+                 1.0) *
+                static_cast<double>(core::CMemory::dynamicSize(m_ItemCache)));
```

Comment:
On reflection, this isn't quire right either. `core::CMemory::dynamicSize(m_ItemCache)` will include all the (key, value) pair static memory, but we're only really adding buckets at this point. I think this should be:
```
memory += static_cast<std::size_t>(
                (static_cast<double>(this->nextItemCacheBucketCount()) /
                     static_cast<double>(m_ItemCache.bucket_count()) -
                 1.0) *
                static_cast<double>(core::CMemory::dynamicSize(m_ItemCache) - m_ItemCache.size() * sizeof(typename TCompressedKeyCacheItemUMap::value_type)));
```